price/pricing/pricer.go

```diff
@@ -0,0 +1,210 @@
+package pricing
+
+import (
+	"fmt"
+	"math/big"
+	"sync"
+	"time"
+
+	"github.com/fln/pprotect"
+
+	"github.com/mysteriumnetwork/payments/crypto"
+	"github.com/rs/zerolog/log"
+)
+
+type Bound struct {
+	Min, Max float64
+}
+
+type FiatPriceAPI interface {
+	MystUSD() float64
+}
+
+type Pricer struct {
+	priceAPI      FiatPriceAPI
+	priceLifetime time.Duration
+	mystBound     Bound
+
+	lock        sync.Mutex
+	lp          LatestPrices
+	cfgProvider ConfigProvider
+}
+
+func NewPricer(cfgProvider ConfigProvider, priceAPI FiatPriceAPI, priceLifetime time.Duration, sensibleMystBound Bound) (*Pricer, error) {
+	pricer := &Pricer{
+		cfgProvider:   cfgProvider,
+		priceAPI:      priceAPI,
+		priceLifetime: priceLifetime,
+		mystBound:     sensibleMystBound,
+	}
+	pprotect.CallLoop(func() {
```

Comment:
What is the purpose of this call loop with no panic handlers and this not being a new goroutine? Do you expect something to panic that Im missing? If so can `pricer.init` also panic as it's doing the same thing? Or is this just leftover code?

Reply:
@mdomasevicius ^ :)

Reply:
sorry left a :hankey: